source/features/user-profile-follower-badge.js

```diff
@@ -0,0 +1,23 @@
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import {getUsername} from '../libs/utils';
+import api from '../libs/api';
+import {getCleanPathname} from '../libs/page-detect';
+
+export default async function () {
+	const username = getUsername();
+	const currentProfile = getCleanPathname();
+	const badge = <div class="follower-badge">Follows you</div>;
+
+	if (currentProfile === username) {
+		return;
+	}
+
+	try {
+		await api(`users/${currentProfile}/following/${username}`, {accept404: true});
+	} catch (error) {
+		if (error.message.includes('JSON')) {
+			select('.vcard-names-container.py-3.js-sticky.js-user-profile-sticky-fields').after(badge);
+		}
+	}
```

Comment:
accept404 only prevents the console.error() in api.js when a user does not follow you. 204 No Content is a successful response, which throws a TypeError from the JSON parse() call in api.js.

Reply:
I double checked and this indeed isn't right. Our `accept404` code is never reached because `.json()` _always_ fails on empty responses.

I found this discussion on the topic but it's pretty unhelpful: https://github.com/whatwg/fetch/issues/113

I think it should be fixed this way in `api.js`:

```js
	const content = await response.text();
	const json = content.length > 0 ? JSON.parse(content) : {};
// 	const json = await response.json();
```

Reply:
Hmm I'm not sure I follow. 404 responses have a message body, so the `.json()` call doesn't fail.

Reply:
Weird, their documentation doesn't mention that. But the suggested fix still stands since `204` responses aren't cached.

Reply:
In short I think `api()` should never throw an error. "Expected" responses should be handled+cached in `api()`

Reply:
👍